### PR TITLE
Support yaml document streams and json streams

### LINKED ISSUE #5: Feature request: `YAML stream` format
Hi, thanks for maintaining a great project!

Would you mind if I contributed this feature?

Context:

I wanted to use faq for manipulating a YAML stream, as well as I use jq for manipulating NDJSON(Or line-delimited JSON).

A YAML stream would look like this. It is basically multiple YAML docs separated by `---`:

```
$ jsonnet -y /dev/stdin <<< '[{x: 1, y: self.x} + {x: 10}, {z: 2}]'
---
{
   "x": 10,
   "y": 10
}
---
{
   "z": 2
}
...
```

Current it doesn't work, obviously, as faq supports a YAML array in a single YAML file only, rather than a YAML stream:

```
$ jsonnet -y /dev/stdin <<< '[{x: 1, y: self.x} + {x: 10}, {z: 2}]' | faq
Error: no supported format found named auto
Usage:
  faq [flags] [filter string] [files...]

Flags:
  -a, --ascii-output        force output to be ascii instead of UTF-8
  -C, --color-output        colorize the output (default true)
  -c, --compact             compact instead of pretty-printed output
  -f, --format string       input format (default "auto")
  -h, --help                help for faq
  -m, --maintain-format     maintain original format (don't output JSON)
  -M, --monochrome-output   monochrome (don't colorize the output)
  -r, --raw                 output raw strings, not JSON texts
  -S, --sort-keys           sort keys of objects on output
  -t, --tab                 use tabs for indentation
```

Note that the error message `Error: no supported format found named auto` is rather confusing as it doesn't correctly express the fact that faq was just unable to auto-detect the format.
I'll open an another dedicated issue for that!

### PR DESCRIPTION
**Pull request by chancez**
Support multiple yaml documents in a single file/stream separated by `---`

Use go-yaml Decoder to handle document streams and copy ghodss/yaml
convertToJSONableObject to formats/yaml_helpers.go to enable converting
yaml to json.

Replaces #24
Closes #5 

Also supports JSON streams.

### CODE REVIEW

File: formats/formats.go [hunk 1]
```diff
@@ -9,6 +9,16 @@ type Encoding interface {
 	Color([]byte) ([]byte, error)
 }
 
+type ToJSONDecoder interface {
+	MarshalJSONBytes() ([]byte, error)
+}
+
+// Streamable represents any format that is decodable as a stream and
+// Isomorphic with JSON.
+type Streamable interface {
```
Comment (chancez) [inline formats/formats.go]: naming is hard

**Comment by chancez** [comment]
Oops forgot to update the slurp section. Fixing now.

File: formats/formats.go [hunk 2]
```diff
@@ -9,6 +9,16 @@ type Encoding interface {
 	Color([]byte) ([]byte, error)
 }
 
+type ToJSONDecoder interface {
```
Comment (jzelinskie) [inline formats/formats.go]: doc comment

File: formats/formats.go [hunk 3]
```diff
@@ -9,6 +9,16 @@ type Encoding interface {
...
+	MarshalJSONBytes() ([]byte, error)
+}
+
+// Streamable represents any format that is decodable as a stream and
+// Isomorphic with JSON.
```
Comment (jzelinskie) [inline formats/formats.go]: no need to capitalize isomorphic

File: formats/json.go [hunk 4]
```diff
@@ -46,6 +46,29 @@ func (jsonEncoding) Color(jsonBytes []byte) ([]byte, error) {
...
+func (jsonEncoding) NewDecoder(jsonBytes []byte) ToJSONDecoder {
+	decoder := json.NewDecoder(bytes.NewBuffer(jsonBytes))
+	return &jsonDecoder{decoder}
+}
+
+type jsonDecoder struct {
+	decoder *json.Decoder
+}
+
+func (decoder *jsonDecoder) MarshalJSONBytes() ([]byte, error) {
```
Comment (jzelinskie) [inline formats/json.go]: just use `d` instead of `decoder` as the receiver

File: formats/yaml.go [hunk 5]
```diff
@@ -2,12 +2,15 @@ package formats
...
+	"encoding/json"
 
 	"github.com/alecthomas/chroma/quick"
 	"github.com/ghodss/yaml"
+	goyaml "gopkg.in/yaml.v2"
 )
 
-type yamlEncoding struct{}
+type yamlEncoding struct {
+}
```
Comment (jzelinskie) [inline formats/yaml.go]: this only needs to be one line

**Comment by chancez** [comment]
Squashed and fixed the review comments.

**Pull request merged**